src/argilla/client/users.py

```diff
@@ -0,0 +1,310 @@
+    """
+
+    __client: "httpx.Client"
+    username: str
+    id: UUID
+    first_name: str
+    last_name: Optional[str]
+    full_name: Optional[str]
+    role: UserRole
+    workspaces: Optional[List[str]]
+    api_key: str
+    inserted_at: datetime
+    updated_at: datetime
+
+    def __init__(
+        self,
+        name: Optional[str] = None,
+        *,
+        id: Optional[Union[str, UUID]] = None,
+    ) -> None:
+        """Doesn't initialize the `User`, as it needs to be instantiated via any of
+        the following classmethods, as the Argilla API is required.
+            - `User.create` to create a new user in Argilla
+            - `User.from_name` to get an existing user from Argilla by its name
+            - `User.from_id` to get an existing user from Argilla by its ID
+            - `User.list` to list all the users in Argilla
+
+        Args:
+            name: the name of the user to be managed. Defaults to None.
+            id: the ID of the user to be managed. Defaults to None.
+
+        Raises:
+            Exception: if the `User` is initialized via the `__init__` method.
+        """
+        error_msg = (
+            "`User` cannot be initialized via the `__init__` method, as it needs"
+            " to send requests to the Argilla API, so it can just be instantiated via"
+            " the following classmethods: `User.create`, `User.from_name`,"
+            " `User.from_id` and `User.list`; note that it must exist in"
+            " advance in Argilla, unless you are creating a new one."
+        )
+        if id is not None:
+            error_msg += f" As the `id` argument is not None, you should use `User.from_id('{id}')` instead."
+        if name is not None:
+            error_msg += f" As the `name` argument is not None, you should use `User.from_name('{name}')` instead."
+        raise Exception(error_msg)
+
+    def __str__(self) -> str:
+        return (
+            f"User(id={self.id}, username={self.username}, role={self.role},"
+            f" workspaces={self.workspaces}, api_key={self.api_key},"
+            f" first_name={self.first_name}, last_name={self.last_name}, role={self.role},"
+            f" inserted_at={self.inserted_at}, updated_at={self.updated_at})"
+        )
+
+    @staticmethod
+    def __active_client(httpx: bool = True) -> Union["httpx.Client", "AuthenticatedClient"]:
+        """Returns the active Argilla `httpx.Client` instance."""
+        try:
+            client = active_client().http_client
+            if httpx:
+                client = client.httpx
+            return client
+        except Exception as e:
+            raise RuntimeError(f"The `rg.active_client()` is not available or not respoding.") from e
```

Comment:
I think that we should create a base class that could be used in client classes that have to use the `httpx` client.